lib/Domain/StyleFixer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Domain;
+
+final class StyleFixer
+{
+    /**
+     * @var StyleProposer[]
+     */
+    private $propsers;
+
+    public function __construct(StyleProposer ...$propsers)
+    {
+        $this->propsers = $propsers;
+    }
+
+    public function fix(string $code): string
+    {
+        foreach ($this->propsers as $proposer) {
+            $code = $proposer->propose($code)->apply($code);
+        }
+
+        return $code;
+    }
+
+    public function fixIntersection(string $code, TextEdits $previouslyAppliedChanges): string
+    {
+        foreach ($this->propsers as $proposer) {
+            $code = $proposer
+                ->propose($code)
+                ->intersection($previouslyAppliedChanges->appliedTextEdits())
```

Comment:
In order for this to work we need to have the text edit ranges _after_ they have been applied. `appliedTextEdits` modifies the start offset of the text edits as if they had been applied.